### PR TITLE
Make JDK HttpClient customizable

### LINKED ISSUE #999: Support customization of JDK 11+ HttpClient
**Is your feature request related to a problem? Please describe.**
Currently I need to add Proxy Information to JDK 11+ HttpClient.

**Describe the solution you'd like**
Exposing a `HttpClient.Builder`

**Describe alternatives you've considered**
Creating a `@Bean` and adding `@Primary`

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by driverpt**
Fixes #999

### CODE REVIEW

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/clientconfig/Http2ClientFeignConfiguration.java [hunk 1]
```diff
@@ -18,8 +18,11 @@
 
 import java.net.http.HttpClient;
 import java.time.Duration;
+import java.util.List;
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/clientconfig/Http2ClientFeignConfiguration.java:21]: Please update the year in the license to `2013-2024`.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/clientconfig/Http2ClientFeignConfiguration.java [hunk 2]
```diff
@@ -34,12 +37,19 @@
...
-				.followRedirects(httpClientProperties.isFollowRedirects() ? HttpClient.Redirect.ALWAYS
-						: HttpClient.Redirect.NEVER)
-				.version(HttpClient.Version.valueOf(httpClientProperties.getHttp2().getVersion()))
-				.connectTimeout(Duration.ofMillis(httpClientProperties.getConnectionTimeout())).build();
+			.followRedirects(httpClientProperties.isFollowRedirects() ? HttpClient.Redirect.ALWAYS
+				: HttpClient.Redirect.NEVER)
+			.version(HttpClient.Version.valueOf(httpClientProperties.getHttp2().getVersion()))
+			.connectTimeout(Duration.ofMillis(httpClientProperties.getConnectionTimeout()));
+	}
+	@Bean
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/clientconfig/Http2ClientFeignConfiguration.java:48]: Please add empty line between methods.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/clientconfig/http2client/Http2ClientCustomizer.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.openfeign.clientconfig.http2client;
+
+import java.net.http.HttpClient;
+
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/clientconfig/http2client/Http2ClientCustomizer.java:19]: Please add javadocs for class and method.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello @driverpt, thanks for creating the PR. Please address the comments. Also, please add a test (you can use Spring Boot's `ApplicationContextRunner`).

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Thanks @driverpt . LGTM. Just one more cosmetic things in the javadocs - please address.

**Review by OlgaMaciaszek — APPROVED** [review]
Thanks @driverpt. LGTM.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Acutally, there seem to be checkstyle issues. Please fix.

**Comment by driverpt** [comment]
Done!

**Review by OlgaMaciaszek — APPROVED** [review]
LGTM.

**Pull request merged**